src/zope/app/tree/node.py

```diff
@@ -89,12 +97,12 @@ def getChildObjects(self):
         if filter:
             return [child for child in children if filter.matches(child)]
         return children
-        
+
     def getChildNodes(self):
         """See zope.app.tree.interfaces.INode"""
         if not self.expanded:
             return []
-        if not hasattr(self, '_child_nodes'):
+        if self._child_nodes is _MARKER:
```

Comment:
I originally had test failures when I used `None`...but I'm not reproducing them now :\